src/java.desktop/share/classes/javax/swing/plaf/synth/SynthGraphicsUtils.java

```diff
@@ -665,9 +670,17 @@ static void paintText(Graphics g, SynthMenuItemLayoutHelper lh,
                 g.setColor(lh.getStyle().getColor(
                         lh.getContext(), ColorType.TEXT_FOREGROUND));
                 g.setFont(lh.getStyle().getFont(lh.getContext()));
+
+                int mnemIndex = lh.getMenuItem().getDisplayedMnemonicIndex();
+                // Check to see if the Mnemonic should be rendered in GTK.
+                if (UIManager.getBoolean("RootPane.altPress")
```

Comment:
You may add another condition to the `if` statement, `mnemIndex < 0` — if there's no mnemonic defined, there's no need to query the properties from `UIManager` at all.

Is the new property even needed? Is `Button.showMnemonics` not enough? This property controls mnemonics on all the components, including menu bar, in Windows and Aqua. Can't Synth and GTK follow the same pattern?

Reply:
> You may add another condition to the if statement, mnemIndex < 0 — if there's no mnemonic defined, there's no need to query the properties from UIManager at all.

Updated.